### PR TITLE
enhance hg pushlog with link to buglist on bugzilla 

### PR DESCRIPTION
**Pull request by philipp-sumo**
this is a proposal for an enhancement for the addon.
it would add a link to the top right of a hg pushlog url providing a link to bugzilla with the list of bugs that get mentioned in that range of patches (on bugzilla you'd then have the benefit of sorting by component, author of the patch or further filtering...)

### CODE REVIEW

File: buglist-link.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+var bugs = [];
```
Comment (calixteman) [inline buglist-link.js]: it could be const and insided getCommonLandings

File: buglist-link.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+function getCommonLandings() {
+	//look for Bug # on page
+	let result;
+	let re = /Bug ([0-9]+)/ig;
+	while ((result = re.exec(document.body.innerText)) !== null) {
+		bugs.push(result[1]);
+	}
+	//consolidate Bug #
+	bugs = new Set(bugs);
```
Comment (calixteman) [inline buglist-link.js]: bugs should be a Set instead of a list at definition, and use add instead of push

File: buglist-link.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	a.textContent = 'View as Buglist on Bugzilla';
+	a.href = 'https://bugzilla.mozilla.org/buglist.cgi?bug_id=';
+	a.id = 'addon-buglist';
+	a.style = 'float:right';
+
+	//create link to buglist
+	titlebar.appendChild(a);
+	
+	//fill link to buglist with data
+	for (let bug of bugs) {
```
Comment (calixteman) [inline buglist-link.js]: you could generate url before to create ``a``:
``` javascript
let url = "https://bugzilla.mozilla.org/buglist.cgi?bug_id=";
bugs.forEach(bug => {
    url += bug + ",";
});
```
and then ``a.href = url;``

File: buglist-link.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		bugs.push(result[1]);
+	}
+	//consolidate Bug #
+	bugs = new Set(bugs);
+	
+	if (bugs.size === 0) {
+		return;
+	}
+
+	var title = document.getElementsByClassName('title');
```
Comment (calixteman) [inline buglist-link.js]: ``const titlebar = document.querySelector("div.title");`` gets only the first one and if it's null then return

File: buglist-link.js (hunk 1 above)
Comment (philipp-sumo) [reply buglist-link.js]: i incorporated your feedback in a new commit

**Review by calixteman — APPROVED** [review]
lgtm
Thanks

**Pull request merged**